### PR TITLE
[NONMODULAR] Ports Arcing APCs from paradise.

### PR DESCRIPTION
**Pull request by FlamingLily**
https://github.com/ParadiseSS13/Paradise/blob/365fc87dbfbeb11f73485547849db2b99b85cde6/code/modules/power/apc.dm#L1207
## About The Pull Request
Does what it says on the tin.

When an APC has 2.5 MW in excess, it has a 5% chance to shoot one living mob (any living mob, including cyborgs, and simplemobs) in range with an electrical arc that deals 5-25 damage, every process() tick, which, as I've been told, occurs every two seconds (therefore, every 8 seconds in skyrat time).
This chance increases to 10% at 5MW, and 15% at 7.5MW

Also adds the variable "shock_proof" to APCs. This defaults to FALSE, and does exactly what you'd expect. If set to TRUE, it will prevent the APC from arcing.

APCs can be upgraded to be shock proof with a single sheet of bronze, and can be un-shock-proofed with a wrench.

## Why It's Good For The Game
Adds a bit more depth to engineering's job. SMES are now actually important, as opposed to being able to be bypassed by hotwiring. Super-fuck-shit-superreactors that produce the power of a small star carry a little more danger to them, if handled poorly

And it's just fun for the station to momentarily panic as APCs begin zapping people, for like, 30 seconds until engineering fixes it, which is easy fixed. And then the dust settles and all the station pets are dead.

### Why I'm PRing this down here, and not TG
This one's simple, I'm PRing this down here, as I personally feel this change would not be well received at TG, as it would interact with the balance of the game, and I have no idea how the average TG round plays, so, me touching their balance is a bad idea.

If a maintainer wishes for me to PR it up there, I will do so, but do know that I am 100% throwing you under the bus by saying "I PRed this downstream and was told to bring it up here, so here you go"

## Changelog
:cl:
add: NanoTrasen have modified their APC designs, they're now approximately 0.1% cheaper to create! Unfortunately, they also have a flaw in that when overcharged, they are prone to arcing.
add: Luckily, the ingenious engineers of the frontiers have found what was removed! A single bronze panel on an APC will make them arc-proof once more!
/:cl:

### CODE REVIEW

**Comment by GoldenAlpharex** [comment]
I'm still not really a fan, but whatever I guess.

**Comment by unit0016** [comment]
Very cool, looks solid codewise. May I suggest some way to upgrade them to be shockproof like we can for cameras, though?

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -1363,6 +1365,23 @@
 			charging = APC_NOT_CHARGING
 			chargecount = 0
 
+		if(excess >= 2500000 && !shock_proof) // SKYRAT EDIT ADD - APC Arcing
+			var/shock_chance = 5 // 5%
+			if(excess >= 7500000)
+				shock_chance = 15
+			else if(excess >= 5000000)
```
Comment (GoldenAlpharex) [inline code/modules/power/apc.dm]: Those numbers should be in defines, to help with code readability and ease of modification in the future if needed.

**Comment by FlamingLily** [comment]
> 
> 
> Very cool, looks solid codewise. May I suggest some way to upgrade them to be shockproof like we can for cameras, though?

Cool idea, any ideas on what could be used?

**Comment by unit0016** [comment]
> Cool idea, any ideas on what could be used?

Maybe a sheet of gold? Or, alternatively, to make it harder on engineering - Brass/Bronze whatever the stuff you get when you splash copper onto Iron in-game.

**Comment by DuffCreeper** [comment]
Dang how will I increase my kill count of assistants now without getting lynched :^)

Jokes aside i'm fine with this despite being one of the engineers that love seeing the funny supply number go up

**Comment by FlamingLily** [comment]
> 
> 
> Dang how will I increase my kill count of assistants now without getting lynched :^)
> 
> Jokes aside i'm fine with this despite being one of the engineers that love seeing the funny supply number go up

It's easy, just hotwire the maint grilles themselves to the engine :)

**Comment by Gandalf2k15** [comment]
ok

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -1363,6 +1374,23 @@
...
+				shock_chance = 10
+			if(prob(shock_chance))
+				var/list/shock_mobs = list()
+				for(var/creature in view(get_turf(src), 5)) //We only want to shock a single random mob in range, not every one.
+					if(isliving(creature))
+						shock_mobs += creature
+				if(shock_mobs.len)
+					var/mob/living/living_target = pick(shock_mobs)
+					living_target.electrocute_act(rand(5, 25), "electrical arc")
+					playsound(get_turf(living_target), 'sound/machines/defib_zap.ogg', 75, TRUE) //Man the defib zap sound is cool and also I couldn't find the proper sound.
```
Comment (Gandalf2k15) [inline code/modules/power/apc.dm]: I'd prefer this was an electrical arcing sound https://youtu.be/_2LpCdhuOyQ?t=48

Comment (FlamingLily) [reply code/modules/power/apc.dm]: I found the actual arcing sound and it uses that now

**Review by Gandalf2k15 — APPROVED** [review]
Lgtm

**Comment by Vizoplez** [comment]
This may break my previous PR that deliberately uncapped shocks from it, just make it scale with power in the grid - like how my PR did it; just use a higher divisor.

I'll fix it anyway

**Comment by FlamingLily** [comment]
> 
> 
> This may break my previous PR that deliberately uncapped shocks from it, just make it scale with power in the grid - like how my PR did it; just use a higher divisor.
> 
> I'll fix it anyway

Which PR? I could have a look

**Comment by Gandalf2k15** [comment]
Seems to work OK.

**Pull request merged**